src/neo/SmartContract/ApplicationEngine.Iterator.cs

```diff
@@ -38,6 +44,23 @@ internal protected static bool IteratorNext(IIterator iterator)
             return iterator.Next();
         }
 
+        /// <summary>
+        /// The implementation of System.Iterator.Count.
+        /// Returns the number of entries in the collection.
+        /// </summary>
+        /// <param name="iterator">The iterator to be counted.</param>
+        /// <returns>Seek the iterator to the last entry and return the number of entries.</returns>
+        internal protected int IteratorCount(IIterator iterator)
+        {
+            int count = 0;
+            while (iterator.Next())
```

Comment:
Won't this change the iterator position?

Reply:
Yes

Reply:
Is this by design, to change iterator position?